kafka-scheduler/src/main/java/com/mesosphere/dcos/kafka/scheduler/KafkaScheduler.java

```diff
@@ -120,18 +125,28 @@ public KafkaScheduler(KafkaSchedulerConfiguration configuration, Environment env
 
         List<Phase> phases = Arrays.asList(
                 ReconciliationPhase.create(reconciler),
-                KafkaUpdatePhase.create(
+                new KafkaUpdatePhase(
                         configState.getTargetName().toString(),
                         envConfig,
                         frameworkState,
-                        offerRequirementProvider));
+                        offerRequirementProvider,
+                        getPhaseStrategyFactory(envConfig)));
+
+        //TODO(Mehmet): CanaryStrategy does not work!! There is no way to send continue/interrupt to a Phase
+
         // If config validation had errors, expose them via the Stage.
         this.installPlan = stageErrors.isEmpty()
-                ? new DefaultPlan("deploy", phases, getPhaseStrategyFactory(envConfig))
-                : new DefaultPlan("deploy", phases, getPhaseStrategyFactory(envConfig),stageErrors);
+                ? new DefaultPlan("deploy", phases, new SerialStrategy<>())
```

Comment:
This could always call `DefaultPlan(String, List<Phase>, Strategy<Phase>, List<String>)`, and just pass an empty list of errors when there are no errors.